src/types.rs

```diff
@@ -568,14 +568,27 @@ fn base64(data: &impl Serialize) -> Result<String, serde_json::Error> {
 
 /// An ACME authorization's state as described in RFC 8555 (section 7.1.4)
 #[derive(Debug, Deserialize)]
+#[non_exhaustive]
 #[serde(rename_all = "camelCase")]
 pub struct AuthorizationState {
     /// The identifier that the account is authorized to represent
-    pub identifier: Identifier,
+    identifier: Identifier,
     /// Current state of the authorization
     pub status: AuthorizationStatus,
     /// Possible challenges for the authorization
     pub challenges: Vec<Challenge>,
+    /// Whether the identifier represents a wildcard domain name
+    #[serde(default)]
+    pub wildcard: bool,
+}
+
+impl AuthorizationState {
+    /// Creates a string for the identifier in this authorization
+    ///
+    /// Takes the `wildcard` field into account.
```

Comment:
I'm not sure this line adds a lot of value. It might be better if it said _how_ it's taken into account, but I think that's already covered by `Identifier::authorized()`.